ccm_web/server/src/canvas/canvas.interfaces.ts

```diff
@@ -1,8 +1,14 @@
-export interface OAuthResponseQuery {
+export interface OAuthGoodResponseQuery {
   code: string
   state: string
 }
 
+export interface OAuthErrorResponseQuery {
+  error: string
+  error_description: string
+}
+
+export interface OAuthResponseQuery extends OAuthGoodResponseQuery, OAuthErrorResponseQuery {}
```

Comment:
So, what's the expected shape here? Is it `OAuthGoodResponseQuery` **or** `OAuthErrorQuery`, or both combined? If it's the former, I think we want to change the type on line 47 of the controller so it's `OAuthGoodResponseQuery | OAuthErrorQuery`. Then we probably want a type guard to check for `error` and `error_description`, something like this?

```
function isOAuthErrorResponseQuery (data: Record<string, unknown>): data is OAuthErrorResponseQuery {
  return 'error' in data && 'error_description' in data
}
```

Then you can change the type check on 53 of the controller, to `isOAuthErrorResponseQuery(query)`.

Maybe this is change is marginal, but I'd like to ensure the types are describing the right objects in the two scenarios.

Reply:
Thanks! I resolved this as per the suggestion. I also added `canvas.utils.ts` file  to simplify the canvas controller file a bit. LMK what you think about this.